### PR TITLE
FIx PR #17 and MNT: rerender

### PR DESCRIPTION
**Pull request by mikemhenry**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mikemhenry** [comment]
@hmaarrfk :wave: working on fixing some issues with pytorch packages that don't seem to pin the pytorch version, which then causes some problems: see https://github.com/pyg-team/pytorch_geometric/issues/3660

So I'm working to add some pytorch version pins to this package:

https://github.com/conda-forge/pytorch_geometric-feedstock/pull/20/files#diff-ff61408cdc05bc9667deeadb55e4aaceb1371972076b6bf6934f9008920f2bd2R5

https://github.com/conda-forge/pytorch_geometric-feedstock/pull/20/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR37

But I'm getting an error 
`OSError: /home/conda/feedstock_root/build_artifacts/pytorch_geometric_1642806414629/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.9/site-packages/torch_sparse/_convert_cpu.so: undefined symbol: _ZNK2at6Tensor6deviceEv`

And it looks like `pytorch_sparse` also doesn't have `pytorch` versions pinned/a matrix setup to support multiple versions:

https://github.com/conda-forge/pytorch_sparse-feedstock/blob/master/recipe/meta.yaml

So do you know if I will need to make the same fix to that package as well?

Thanks!
@jaimergp  you might also know the answer to this question

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ requirements:
     - scikit-learn
     - scipy
     - tqdm
-    - pytorch
+    - pytorch {{ pytorch_version }}
```
Comment (hmaarrfk) [inline recipe/meta.yaml:37]:
```suggestion
    - pytorch 1.10
```

I think we made a mistake a pinned to tightly. I would suggest loosening this one

**Comment by hmaarrfk** [comment]
This is a noarch package, which package is broken? We recently introduced a migrator and the recent PR for 1.10.1 should help alleviate future pinning problems.

**Comment by hmaarrfk** [comment]
Torchsparse seems broken.

**Comment by hmaarrfk** [comment]
it should be rebuilt.

**Comment by mikemhenry** [comment]
Sorry yes torchsparse was the one that was then giving me some ABI/symbols, looking at the package it didn't pin pytorch hard enough

**Comment by hmaarrfk** [comment]
@conda-forge-admin please restart cis

**Comment by hmaarrfk** [comment]
hmmm. i'm really puzzled...

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -1,5 +1,2 @@
 pytorch_version:
-  - 1.8.0
-  - 1.9.0
-  - 1.9.1
```
Comment (hmaarrfk) [inline recipe/conda_build_config.yaml:4]: Can you delete this file and pull in the migration file manually?

Comment (mikemhenry) [reply recipe/conda_build_config.yaml:4]: Sure, which one should I pull in? The torchvision feedstock had a lot and I'm not sure which one is needed here.

Comment (hmaarrfk) [reply recipe/conda_build_config.yaml:4]: This file is the one you need to copy over

https://github.com/conda-forge/torchvision-feedstock/blob/master/.ci_support/migrations/pytorch110.yaml

Comment (mikemhenry) [reply recipe/conda_build_config.yaml:4]: done!

**Comment by mikemhenry** [comment]
@conda-forge-admin, please rerender

**Comment by mikemhenry** [comment]
@conda-forge-admin, please rerender

**Comment by mikemhenry** [comment]
@hmaarrfk trying to get this working again our current error is https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=505349&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=28

`'cuda_compiler_version' is undefined`
Which is surprising?

**Comment by hmaarrfk** [comment]
Are compilers needed? Why is this not a noarch?

**Comment by mikemhenry** [comment]
@hmaarrfk it is https://github.com/conda-forge/pytorch_geometric-feedstock/pull/20/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR17

But that error is coming from https://github.com/conda-forge/pytorch_geometric-feedstock/pull/20/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR4 so we can set `pytorch {{ pytorch_version }} {{ torch_proc_type }}*` correctly

**Comment by hmaarrfk** [comment]
The solvers should already be taking care of this.

You can simply remove the pinning.

**Comment by mikemhenry** [comment]
@hmaarrfk ready for review/merge

**Comment by hmaarrfk** [comment]
Looks good. Did you want to maintain

**Comment by mikemhenry** [comment]
Sure! Thanks @hmaarrfk

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Pull request merged**